src/test/java/page/GoogleSearchForWikipediaResultsPage.java

```diff
@@ -0,0 +1,49 @@
+package page;
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.PageFactory;
+
+import java.util.List;
+
+public class GoogleSearchForWikipediaResultsPage extends BasePage{
+
+    private WebDriver driver;
+    private String searchTerm;
+
+    private String wikiRelatedLocator = "//div[contains(@class, 'g') and contains(., 'wikipedia')]";
+
+    @FindBy(xpath = "//div[contains(@class, 'g')")
+    private List<WebElement> generalGoogleForWikipediaSearchResult;
+
+    @FindBy(xpath = "//*[@id='rso']//a/h3")
+    private WebElement firstWikiLinkFromGoogleSearch;
+
+    public GoogleSearchForWikipediaResultsPage(WebDriver driver, String searchTerm) {
+        this.searchTerm = searchTerm;
+        this.driver = driver;
+        PageFactory.initElements(driver, this);
+    }
+
+    public int countGeneralNumberOfSearchResults() {
+        return generalGoogleForWikipediaSearchResult.size();
+    }
+
+    public String wikipediaPageTitle() {
+                return driver.getTitle();
+    }
+
+    public String searchResultPageUrl(){
+        return driver.getCurrentUrl();
+    }
+
+    public int searchResultLength(){
+        return driver.getTitle().length();
+    }
+
+    public GoogleSearchForWikipediaResultsPage openWikipediaPage() {
```

Comment:
Don't like this method. It is not universal. You need create openWikipediaPage like you did for Google Page

Reply:
I'm doing it WikipediaHomePage. This one is left for testing aims.